polaris-react/src/components/Navigation/Navigation.scss

```diff
@@ -213,13 +221,22 @@ $disabled-fade: 0.6;
   }
 }
 
+.Text-truncated {
+  // stylelint-disable-next-line value-no-vendor-prefix
+  display: -webkit-box;
+  overflow: hidden;
+  -webkit-line-clamp: 1;
```

Comment:
I was suggesting a global line clamp rule instead of using it here with a value of one. Does `@include truncate;` not work here?

Reply:
I wasn't aware of the `@include truncate;` mixin, but after trying it I run into the issue I had originally. So when using the `overflow: hidden;` + `text-overflow: ellipsis;` method (that the @truncate mixin uses) you need to set a specific width on the element being truncated. This becomes an issue because the secondary action needs to be accounted for AND the badge needs to be accounted for, so in my original PR I had that, but if you have both a badge and secondary action it becomes messy and doesn't quite fit. 

The ` -webkit-line-clamp: 1;` method doesn't require a specified width on the element which is convenient because with that we can fit both badges and secondary actions (which does happen in the admin nav).